src/views/AddItem.jsx

```diff
@@ -12,24 +12,56 @@ const dayConverter = (text) => {
 	}
 };
 
-export function AddItem({ listId }) {
+export function AddItem({ listId, data }) {
 	const [itemName, setItemName] = useState('');
 	const [frequency, setFrequency] = useState('soon');
-	const [message, setMessage] = useState('');
+	const [itemMessage, setItemMessage] = useState('');
+	const [errorMessage, setErrorMessage] = useState('');
+
+	const clearMessage = (messageToAdjust) => {
+		setTimeout(() => {
+			messageToAdjust('');
+		}, 3000);
+	};
 
 	const handleSubmit = async (e) => {
 		e.preventDefault();
+
+		if (!itemName.trim()) {
+			setErrorMessage('Please enter item name.');
+			clearMessage(setErrorMessage);
+			return;
+		}
+
+		const existingItem = data.find((item) => {
+			return (
+				itemName.localeCompare(item.name, 'en', {
+					sensitivity: 'base',
+					ignorePunctuation: true,
+				}) === 0
+			);
+		});
+
+		if (existingItem !== undefined) {
+			setErrorMessage(`${existingItem.name} is already in your shopping list.`);
+			clearMessage(setErrorMessage);
```

Comment:
I find this line introduces indirection to the code and I think it would be best to keep it simple.

This is what happens when I read this:
1. I remember that a `clearMessage` function exist, and I know that it "resets" _a_ state.
2. I see that you're passing `setErrorMessage` here, but I wonder why.
3. Turns out, you're passing `setErrorMessage` to `clearMessage` so it resets it.

I suggest a few things to make this better:

1. stop passing `setErrorMessage` and call `setErrorMessage` directly in `clearMessage`
2. Rename `clearMessage` to `clearErrorMessage`, then create another function called `clearItemMessage`

I believe that the intent here is to DRY your code, which is a valid argument, but I find that passing around the `set*` function is a bit odd, especially because `clearMessage` doesn't have the same naming structure as the functions that update state

```js
setTimeout(() => {
  // 1. it doesn't start with `set*`
  // 2. it doesn't describe an action
	messageToAdjust('');
}, 3000);
```

To wrap it up:

1. This feels like the abstraction `clearMessage` went "one level too deep"
2. Having two function that do the same `clearErrorMessage` and `clearItemMessage` is valid. 

If you have time, change it (I know it's Saturday evening already). If you don't, that's fine but, keep it in mind for the future!

Some good videos about this:
- https://www.youtube.com/watch?v=rQlMtztiAoA
- https://www.youtube.com/watch?v=8bZh5LMaSmE&t=951s (this is a **fantastic** talk)

Reply:
Thanks for this thoughtful comment! We have refactored this into two separate functions to be more understandable.

Reply:
Creating two separate clearMessage functions is definitely easier to understand / follow - thank you for making this suggestion!